multisig/tests/multisig_blackbox_test.rs

```diff
@@ -1050,3 +1154,1348 @@ fn multisig_non_board_member_sign_test() {
+            board,
+        )
+        .code(MULTISIG_CODE_PATH)
+        .new_address(MULTISIG_ADDRESS)
+        .run();
+
+    state.mock_multi_transfer_deploy();
+
+    let tx_id = 1u64;
+
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .add_unprocessed_refund_tx_to_batch(tx_id)
+        .run();
+}
+
+#[test]
+fn test_withdraw_slashed_amount_success() {
+    let mut state = MultiTransferTestState::new();
+
+    state.deploy_contracts_config();
+
+    let slashed_amount = BigUint::from(500u64);
+    state.world.set_esdt_balance(
+        MULTISIG_ADDRESS,
+        WEGLD_TOKEN_ID.as_bytes(),
+        slashed_amount.clone(),
+    );
+
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .slash_board_member(RELAYER1_ADDRESS.to_managed_address())
+        .run();
+
+    let remaining_stake = state
+        .world
+        .query()
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .amount_staked(RELAYER1_ADDRESS.to_managed_address())
+        .returns(ReturnsResult)
+        .run();
+
+    assert_eq!(remaining_stake, BigUint::from(500u64));
+
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .withdraw_slashed_amount()
+        .run();
+
+    let remaining_slashed_amount = state
+        .world
+        .query()
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .slash_amount()
+        .returns(ReturnsResult)
+        .run();
+    assert_eq!(remaining_slashed_amount, BigUint::from(500u64));
+}
+
+#[test]
+fn test_withdraw_refund_fees_for_ethereum_success() {
+    let mut state = MultiTransferTestState::new();
+
+    let mut board: MultiValueEncoded<StaticApi, ManagedAddress<StaticApi>> =
+        MultiValueEncoded::new();
+    board.push(ManagedAddress::from(RELAYER1_ADDRESS.eval_to_array()));
+    board.push(ManagedAddress::from(RELAYER2_ADDRESS.eval_to_array()));
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .init(
+            MOCK_ESDT_SAFE_ADDRESS,
+            MULTI_TRANSFER_ADDRESS,
+            BRIDGE_PROXY_ADDRESS,
+            BRIDGED_TOKENS_WRAPPER_ADDRESS,
+            PRICE_AGGREGATOR_ADDRESS,
+            1_000u64,
+            500u64,
+            2usize,
+            board,
+        )
+        .code(MULTISIG_CODE_PATH)
+        .new_address(MULTISIG_ADDRESS)
+        .run();
+    state.mock_safe_deploy();
+
+    state
+        .world
+        .tx()
+        .from(OWNER_ADDRESS)
+        .to(MULTISIG_ADDRESS)
+        .typed(multisig_proxy::MultisigProxy)
+        .withdraw_refund_fees_for_ethereum(TokenIdentifier::from(WEGLD_TOKEN_ID))
+        .run();
```

Comment:
check the balance of the owner

Reply:
we are using a mock esdt safe contract, so after multisig calling withdraw_refund_fees_for_ethereum from esdt safe, nothing is happening. maybe i can use a storage mapper to see we actually get here, otherwise the functionality from  withdraw_refund_fees_for_ethereum on esdt safe was already unit tested in esdt_safe_blackbox_test.rs

Reply:
No need then 🙏